packages/cli/lib/projects.js

```diff
@@ -168,14 +168,51 @@ const validateProjectConfig = (projectConfig, projectDir) => {
   }
 };
 
+const pollFetchProject = async (accountId, projectName) => {
+  // Temporary solution for gating slowness. Retry on 403 statusCode
+  try {
+    const project = await fetchProject(accountId, projectName);
+    return project;
+  } catch (error) {
+    if (error.statusCode === 404) {
+      throw error;
+    }
+  }
+  return new Promise((resolve, reject) => {
+    let pollCount = 0;
+    const spinnies = SpinniesManager.init();
+    spinnies.add('pollFetchProject', {
+      text: 'Fetching project status',
+    });
+    const pollInterval = setInterval(async () => {
+      try {
+        const project = await fetchProject(accountId, projectName);
+        if (project) {
+          spinnies.remove('pollFetchProject');
+          clearInterval(pollInterval);
+          resolve(project);
+        }
+      } catch (err) {
+        // Poll up to max 30s
+        if (err.statusCode === 404 || pollCount >= 15) {
+          spinnies.remove('pollFetchProject');
+          clearInterval(pollInterval);
+          reject(err);
+        }
+        pollCount += 1;
+      }
+    }, POLLING_DELAY);
+  });
+};
+
 const ensureProjectExists = async (
   accountId,
   projectName,
   { forceCreate = false, allowCreate = true, noLogs = false } = {}
 ) => {
   const accountIdentifier = uiAccountDescription(accountId);
   try {
-    const project = await fetchProject(accountId, projectName);
+    const project = await pollFetchProject(accountId, projectName);
```

Comment:
I'm thinking that we should only use this new polling logic for the Dev Mode work. Maybe we can add a new `withPolling` option to this arguments in `ensureProjectExists`. I don't think this issue will have much of an effect on the existing project-related commands.

Reply:
Yep exactly - I made it so behavior for all other usages stayed the same, but I like the prop idea better